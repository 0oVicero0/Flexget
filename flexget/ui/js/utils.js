'use strict';

String.prototype.startsWith = function(prefix) {
  return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}

app.menu = [];
app.routes = [];

function register_menu(href, caption, icon, order) {
  href = '/ui/#' + href;
  app.menu.push({href: href, caption: caption, icon: icon, order: order})
}

function register_route(name, url, controller, template) {
  app.routes.push({
    name: name,
    url: url,
    controller: controller,
    template: template
  });
}