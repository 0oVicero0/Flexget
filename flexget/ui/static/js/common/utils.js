'use strict';

String.prototype.startsWith = function(prefix) {
    return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}