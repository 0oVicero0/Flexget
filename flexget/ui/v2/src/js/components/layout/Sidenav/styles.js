import styled, { css } from 'styled-components';
import theme from 'theme';
import Paper from 'material-ui/Paper';
import List, { ListItem } from 'material-ui/List';
import Button from 'material-ui/Button';

export const colorClass = css`color: ${theme.palette.accent[200]};`;

export const ListWrapper = styled(Paper)`
  background-color: ${theme.palette.accent[900]};
  transition: ${theme.transitions.create(['width', 'visibility'])};
  height: auto;
  width: ${({ open }) => (open ? '100vw' : 0)};
  border-radius: 0;
  display: flex;
  flex-direction: column;
  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};

  ${theme.breakpoints.up('sm')} {
    height: calc(100% - 5rem);
    width: ${({ open }) => (open ? '19rem' : '5rem')};
    position: fixed;
    visibility: visible;
  }
`;

export const Label = styled.p`
  composes: ${colorClass};
  flex: 1;
  text-transform: none;
  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};
  opacity: ${({ open }) => (open ? 1 : 0)};
  transition: ${theme.transitions.create(['visibility', 'opacity'])};
  width: ${({ open }) => (open ? 'auto' : 0)};
`;

export const NavList = styled(List)`
  flex: 1;
`;

export const NavItem = styled(ListItem)`
  padding: 0;
  height: 4.8rem;
  border-left: 3px solid transparent;

  &:hover {
    border-left: 3px solid ${theme.palette.primary[500]};
  }
`;

export const NavButton = styled(Button)`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
  min-width: 5rem;
  padding-top: 0;
  padding-bottom: 0;
`;

export const NavVersion = styled.div`
  display: ${({ hide }) => (hide ? 'none' : 'block')};
`;
