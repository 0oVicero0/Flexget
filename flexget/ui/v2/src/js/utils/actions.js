export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
