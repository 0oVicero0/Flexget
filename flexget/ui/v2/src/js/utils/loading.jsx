import Loadable from 'react-loadable';
import Loading from 'components/common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}

