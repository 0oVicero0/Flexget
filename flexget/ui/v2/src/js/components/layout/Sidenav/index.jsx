import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Button from 'material-ui/Button';
import List, { ListItem } from 'material-ui/List';
import Icon from 'material-ui/Icon';
import Paper from 'material-ui/Paper';
import Version from 'containers/layout/Version';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('SideNav', theme => ({
  listWrapper: {
    backgroundColor: theme.palette.accent[900],
    transition: theme.transitions.create(['width', 'visibility']),
    height: '100%',
    width: 190,
    borderRadius: 0,
    display: 'flex',
    flex: {
      direction: 'column',
    },
    [theme.breakpoints.up('sm')]: {
      width: 190,
    },
  },
  sideNavMini: {
    width: 0,
    visibility: 'hidden',
    [theme.breakpoints.up('sm')]: {
      visibility: 'visible',
      width: 50,
    },
  },
  label: {
    color: theme.palette.accent[200],
    flex: 1,
    textTransform: 'none',
  },
  labelMini: {
    display: 'none',
  },
  icon: {
    color: theme.palette.accent[200],
  },
  list: {
    flex: {
      grow: 1,
    },
  },
  listItem: {
    padding: 0,
    height: 48,
    borderLeftWidth: 3,
    borderLeft: 'solid transparent',
    '&:hover': {
      borderLeftWidth: 3,
      borderLeft: `solid ${theme.palette.primary[500]}`,
    },
  },
  button: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
    minWidth: 50,
  },
  versionHide: {
    display: 'none',
  },
}));

const sideNavItems = [
  {
    link: '/log',
    icon: 'file-text-o',
    label: 'Log',
  },
  {
    link: '/execute',
    icon: 'cog',
    label: 'Execute',
  },
  {
    link: '/config',
    icon: 'pencil',
    label: 'Config',
  },
  {
    link: '/history',
    icon: 'history',
    label: 'History',
  },
  {
    link: '/movies',
    icon: 'film',
    label: 'Movies',
  },
  {
    link: '/pending',
    icon: 'check',
    label: 'Pending',
  },
  {
    link: '/schedule',
    icon: 'calendar',
    label: 'Schedule',
  },
  {
    link: '/seen',
    icon: 'eye',
    label: 'Seen',
  },
  {
    link: '/series',
    icon: 'tv',
    label: 'Series',
  },
  {
    link: '/status',
    icon: 'heartbeat',
    label: 'Status',
  },
];

class SideNav extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    sideBarOpen: PropTypes.bool.isRequired,
  };

  renderNavItems() {
    const { classes, sideBarOpen } = this.props;
    return sideNavItems.map(({ link, icon, label }) => (
      <Link to={link} key={link}>
        <ListItem className={classes.listItem}>
          <Button color="accent" className={classes.button}>
            <Icon className={classNames('fa', `fa-${icon}`, classes.icon)} />
            <p className={
              classNames(
                classes.label,
                { [classes.labelMini]: !sideBarOpen },
              )}
            >
              {label}
            </p>
          </Button>
        </ListItem>
      </Link>
    ));
  }

  render() {
    const { classes, sideBarOpen } = this.props;
    return (
      <Paper
        className={classNames(
          classes.listWrapper,
          { [classes.sideNavMini]: !sideBarOpen },
        )}
        elevation={16}
      >
        <List className={classes.list}>
          { ::this.renderNavItems() }
        </List>
        <div className={classNames({ [classes.versionHide]: !sideBarOpen })}>
          <Version />
        </div>
      </Paper>
    );
  }
}

export default withStyles(styleSheet)(SideNav);
