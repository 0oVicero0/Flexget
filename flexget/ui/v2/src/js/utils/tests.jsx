import React from 'react';
import { MuiThemeProvider } from 'material-ui/styles';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

export function themed(component) {
  return (
    <MuiThemeProvider>
      { component }
    </MuiThemeProvider>
  );
}

export function router(component) {
  return (
    <MemoryRouter>
      { component }
    </MemoryRouter>
  );
}

export function provider(component, state = {}) {
  return (
    <Provider store={mockStore(state)}>
      { component }
    </Provider>
  );
}
