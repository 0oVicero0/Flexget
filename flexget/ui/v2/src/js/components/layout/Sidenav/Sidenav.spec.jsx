import React from 'react';
import renderer from 'react-test-renderer';
import Sidenav from 'components/layout/Sidenav';
import { provider, themed, router } from 'utils/tests';
import fetchMock from 'fetch-mock';

describe('components/layout/Sidenav', () => {
  beforeEach(() => {
    fetchMock
        .get('/api/server/version', {});
  });

  it('renders correctly with sideBarOpen', () => {
    const tree = renderer.create(
      provider(router(themed(<Sidenav sideBarOpen />)), { version: {} }),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds mini classes without sideBarOpen', () => {
    const tree = renderer.create(
      provider(router(themed(<Sidenav sideBarOpen={false} />)), { version: {} }),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
