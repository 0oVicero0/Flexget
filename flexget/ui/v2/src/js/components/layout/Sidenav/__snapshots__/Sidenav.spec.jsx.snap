// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Sidenav adds mini classes without sideBarOpen 1`] = `
<div
  className="MuiPaper-root-2758040674 MuiPaper-shadow16-3841796706 MuiPaper-rounded-418257787 SideNav-listWrapper-513568510 SideNav-sideNavMini-2485013513"
>
  <div
    className="MuiList-root-43412855 MuiList-padding-392119320 SideNav-list-3036263360"
  >
    <a
      href="/log"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-file-text-o SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Log
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/execute"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-cog SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Execute
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/config"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-pencil SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Config
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/history"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-history SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              History
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/movies"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-film SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Movies
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/pending"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-check SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Pending
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/schedule"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-calendar SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Schedule
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/seen"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-eye SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Seen
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/series"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-tv SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Series
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/status"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-heartbeat SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758 SideNav-labelMini-5987815"
            >
              Status
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
  </div>
  <div
    className="SideNav-versionHide-5987815"
  >
    <div
      className="Version-version-3221117096"
    >
      <p
        className="Version-par-2841032265"
      >
        Version Info
      </p>
      <p
        className="Version-par-2841032265"
      >
        Flexget: 
         
         
      </p>
      <p
        className="Version-par-2841032265"
      >
        API: 
      </p>
    </div>
  </div>
</div>
`;

exports[`components/layout/Sidenav renders correctly with sideBarOpen 1`] = `
<div
  className="MuiPaper-root-2758040674 MuiPaper-shadow16-3841796706 MuiPaper-rounded-418257787 SideNav-listWrapper-513568510"
>
  <div
    className="MuiList-root-43412855 MuiList-padding-392119320 SideNav-list-3036263360"
  >
    <a
      href="/log"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-file-text-o SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Log
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/execute"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-cog SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Execute
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/config"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-pencil SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Config
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/history"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-history SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              History
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/movies"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-film SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Movies
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/pending"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-check SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Pending
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/schedule"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-calendar SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Schedule
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/seen"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-eye SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Seen
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/series"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-tv SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Series
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
    <a
      href="/status"
      onClick={[Function]}
    >
      <div
        className="MuiListItem-root-3441122480 MuiListItem-gutters-58642691 MuiListItem-default-132630306 SideNav-listItem-2509608802"
        disabled={false}
      >
        <button
          className="MuiButtonBase-root-3177716317 MuiButton-root-325725483 MuiButton-flatAccent-3482837232 SideNav-button-2847385078"
          disabled={false}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-49836587"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-3402991258 fa fa-heartbeat SideNav-icon-2371623600"
            />
            <p
              className="SideNav-label-470754758"
            >
              Status
            </p>
          </span>
          <span
            className="MuiTouchRipple-root-3868442396"
          />
        </button>
      </div>
    </a>
  </div>
  <div
    className=""
  >
    <div
      className="Version-version-3221117096"
    >
      <p
        className="Version-par-2841032265"
      >
        Version Info
      </p>
      <p
        className="Version-par-2841032265"
      >
        Flexget: 
         
         
      </p>
      <p
        className="Version-par-2841032265"
      >
        API: 
      </p>
    </div>
  </div>
</div>
`;
